include/seqan/index/find_swift.h

```diff
@@ -1017,14 +1017,24 @@ inline void _createHit(
 
     int64_t ndlBegin = lastInc + length(pattern.shape) - diag - height;
     auto const & queryInfix = seqan2::getSequenceByNo(ndlSeqNo, seqan2::indexText(seqan2::needle(pattern)));
-    // can not extend seed past the beginning of the host sequence
+    
     if (ndlBegin + (int64_t) seqan2::beginPosition(queryInfix) < 0LL)
-        return;
+    {
+        std::clamp(ndlBegin, -(int64_t)seqan2::beginPosition(queryInfix), (int64_t) 0);
+        height = lastInc + length(pattern.shape) - diag - ndlBegin;
```

Comment:
Is `lastInc + length(pattern) >= diag + ndlBegin` always true?
If yes, maybe add an assertion, if not height can underflow causing new issues down stream.